recipe-server/normandy/recipes/api/v2/serializers.py

```diff
@@ -154,7 +154,17 @@ def validate_extra_filter_expression(self, value):
         jexl.add_transform('preferenceIsUserSet', lambda x: x)
         jexl.add_transform('preferenceExists', lambda x: x)
 
-        errors = list(jexl.validate(value))
+        try:
+            errors = list(jexl.validate(value))
+        except Exception as exc:
+            # The JEXL parser can occasionally throw exceptions when
+            # called to validate certain invalid inputs.
+            # Catch them and at least indicate the field that failed,
+            # even if we can't explain exactly what the problem was.
+            # See https://github.com/mozilla/normandy/issues/1059.
+            error = 'The JEXL parser failed to validate {}'.format(value)
```

Comment:
Normandy runs with Python 3.6, which means we have access to f-strings. That means you could write this as:

```python
error = f'The JEXL parser failed to validate {value}'
```

which is exactly equivalent.

Note that this is a style change. Since it is not enforced by the linter, you can write the code however you like. I just mention it because f-strings are a new Python feature that not many projects get to use.